src/gen.rs

```diff
@@ -17,38 +21,59 @@ pub struct Cmd {
 pub enum Output {
     /// Rust trait, client bindings, and test harness
     Rust,
+    /// Json representation of contract spec types
+    Json,
 }
 
 #[derive(thiserror::Error, Debug)]
 pub enum Error {
     #[error("generate rust from file: {0}")]
-    GenerateRustFromFile(rust::GenerateFromFileError),
+    GenerateRustFromFile(#[from] rust::GenerateFromFileError),
     #[error("format rust error: {0}")]
-    FormatRustError(syn::Error),
+    FormatRust(#[from] syn::Error),
+    #[error("host")]
+    Host(#[from] HostError),
+    #[error("xdr error: {0}")]
+    Xdr(#[from] soroban_env_host::xdr::Error),
+    #[error("io")]
+    Io(#[from] io::Error),
+    #[error("json error: {0}")]
+    Json(#[from] serde_json::Error),
+    #[error("contractnotfound")]
+    ContractSpecNotFound,
```

Comment:
This code intentionally didn't use the #from attribute because the intent was to map specific error situations and not broad error categories.

Reply:
I removed the uses of #from